go/vt/vtgate/executor_select_test.go

```diff
@@ -3942,6 +3942,37 @@ func TestSelectView(t *testing.T) {
 	utils.MustMatch(t, wantQueries, sbc.Queries)
 }
 
+func TestWarmingReads(t *testing.T) {
+	executor, primary, replica := createExecutorEnvWithPrimaryReplicaConn(100)
+
+	executor.normalize = true
+	session := NewSafeSession(&vtgatepb.Session{TargetString: KsTestUnsharded})
+
+	_, err := executor.Execute(context.Background(), "TestSelect", session, "select age, city from user", map[string]*querypb.BindVariable{})
```

Comment:
Would be nice to modify this test so that it runs through a bunch of different cases which exercise the various case statements you have in `executeWarmingReplicaRead`, and also some tests for negatives like `insert`, `update`, and queries that compose `select` with `insert` and `update`.

Reply:
Would also be good to add tests with trailing comments to see how that lands on the replicas.